### DISCUSSION TITLE
Move Form Builder Plugin To Monorepo

### DISCUSSION THREAD

**Discussion by jacobsfletch** [payloadcms/payload#4466] [Roadmap]
As we move the [Form Builder Plugin](https://github.com/payloadcms/payload/tree/main/packages/plugin-form-builder) to the [Payload Monorepo](https://github.com/payloadcms/payload), we need a place to consolidate any open issues from the [old plugin repo](https://github.com/payloadcms/plugin-form-builder) that are not technically bugs. All other issues will be transferred to the Payload monorepo as expected with the label `plugin: form-builder`. This discussion will act a place to document this process and ensure no work is lost.

I'll go into greater detail into each one of these as soon as possible.

Features:
- [ ] [Add an `upload` field](https://github.com/payloadcms/plugin-form-builder/issues/46)
- [ ] [Support multi-select](https://github.com/payloadcms/plugin-form-builder/issues/58). [Here's a proof of concept](https://github.com/payloadcms/plugin-form-builder/pull/59).
- [ ] [Wire up submission validations](https://github.com/payloadcms/plugin-form-builder/issues/50). See PRs below.
- [ ] [Support custom defaultColumns for Form Submissions collection](https://github.com/payloadcms/plugin-form-builder/issues/56)
- [ ] [Grouped fields](https://github.com/payloadcms/plugin-form-builder/issues/74)
- [ ] [Hidden fields](https://github.com/payloadcms/plugin-form-builder/issues/70)

Doc changes:
- [ ] [Add documentation about how to create form submissions](https://github.com/payloadcms/plugin-form-builder/issues/48)
- [ ] [Add docmentation for front-end implementation](https://github.com/payloadcms/plugin-form-builder/issues/51)

Minor fixes:
- [ ] [Require slug in `Partial<Block>` type](https://github.com/payloadcms/plugin-form-builder/issues/34). [Here's a work in progress](https://github.com/payloadcms/plugin-form-builder/pull/43).
- [ ] [Safely fail `variables.find`](https://github.com/payloadcms/plugin-form-builder/issues/44)

PRs:
- [ ] [Docker](https://github.com/payloadcms/plugin-form-builder/pull/53)
- [ ] [Server-side validations](https://github.com/payloadcms/plugin-form-builder/pull/61)

Open issues that have been (or will be) transferred into the Payload Monorepo (I will link to the exact issues as they are opened):
- [ ] [Localized title](https://github.com/payloadcms/plugin-form-builder/issues/49). Open issue #4470.
- [ ] [Form ID validation](https://github.com/payloadcms/plugin-form-builder/issues/52). Open issue #4471.
- [ ] [Handle payment method differs from docs](https://github.com/payloadcms/plugin-form-builder/issues/62). Open issue #4472.
- [ ] [Type error when adding fields to existing block](https://github.com/payloadcms/plugin-form-builder/issues/66). Open issue #4473.
- [ ] [Localized email](https://github.com/payloadcms/plugin-form-builder/issues/69). Open issue #4474.
- [x] [Support Lexical RTE](https://github.com/payloadcms/plugin-form-builder/issues/73). Open issue #4475.
- [ ] 2.0 peer deps. Open issue #4476.

All other open issues from this repo have been (or will be) closed. All new issues and PRs can be made directly in the [Payload Monorepo](https://github.com/payloadcms/payload).

Feel free to leave a comment below with new feature requests, and I'll add them to this list!

**Comment by dominikzogg** [comment]
"Support multi-select. Here's a proof of concept." => new PR https://github.com/payloadcms/payload/pull/4482

**Comment by jacobsfletch** [comment]
Quick update: there's now a test suite setup for this plugin along with some basic integration tests to get us started: https://github.com/payloadcms/payload/pull/4500

**Comment by jacobsfletch** [comment]
Another update: docs have been improved and moved into the `/docs` directory so they'll now surface on the website: #4512
